src/Unirest/Request.php

```diff
@@ -612,15 +603,15 @@ public static function send($method, $url, $body = null, $headers = array(), $us
             }
 
             // if retries are enabled at global and request level
-            if (self::$enableRetries && self::$retryCurrentRequest !== false) {
+            if (self::$enableRetries && self::$overrideRetryForNextRequest !== OverrideRetry::DISABLE_RETRY) {
```

Comment:
Rather than checking what the value is **NOT**, check what it is and apply the conditional behavior on those values. This guards you from bugs in the future in case the enum changes:

```suggestion
            if (self::$enableRetries && (self::$overrideRetryForNextRequest === OverrideRetry::ENABLE_RETRY || self::$overrideRetryForNextRequest === OverrideRetry::USE_GLOBAL_SETTINGS) {
```

Also, I would recommend abstracting out this condition as a method like 	`shouldRetryRequest`. Makes it easier to read the code.

Reply:
Fixed!